docs/docsite/rst/network_aggregate_resources.rst

```diff
@@ -0,0 +1,200 @@
+
+
+.. warning:: Why does ``purge`` default to ``no``?
+
+To prevent accidental deletion, ``purge`` is always set to ``no``. This requires that playbook writers explicitly add ``purge: yes`` to enable this to opt-in to potentially dangerious behaviour.
+
+When would you use aggregate resources with ``purge: true``?
+------------------------------------------------------------
+
+* Ansible to execute your "single source of truth" (execute here means we can talk to source  of truth, e.g. CMS or external data source)
+* Ansible is your "Source of Truth"
+
+
+When would you aggregate resources with ``purge: false``?
+---------------------------------------------------------
+
+
+The *additive* format can be useful in a number of cases:
+
+* When Ansible isn't executing your Single Source of truth; and therefore doesn't ...
+* Allows you to start using Ansible to configure just one part of your network
+* FIXME
+
+
+Local overrides of global module values
+=======================================
+
+When writing tasks using ``aggregate`` you may find yourself repeating various settings within the aggregate dictionary. For example:
+
+.. code-block:: yaml
+
+   - name: Reserve mgmt vlans
+     net_vlan:
+       aggregate:
+         - { vlan_id: 4, name: reserved_vlan, state: active }
+         - { vlan_id: 5, name: mgmt, state: active }
+         - { vlan_id: 6, name: reserved_vlan , state: active}
+       name: reserved_vlan
+       state: active # override
+
+In the above example we can see that ``state: active`` is set for all vlans, and most have ``name: reserved_vlan``. We can simplify this to:
+
+
+.. code-block:: yaml
+
+   - name: Reserve mgmt vlans
+     net_vlan:
+       aggregate:
+         - { vlan_id: 4 }
+         - { vlan_id: 5, name: mgmt }
+         - { vlan_id: 6 }
+       name: reserved_vlan # override
+       state: active # override
+
+
+Note that:
+
+* Shorter task
+* The special cases, ``name: mgmt``, stand out a lot more, increasing readability
+* This can be very powerful when a module uses take a lot of options, most of which are common, such as ``net_interfaces`` or ``nxos_bgp_neighbor_af``
+
+FIXME: Become realy power on ``net_interfaces``, mtu, admin_state, description
+
+
+Reference ID shorthand
+======================
+
+
+.. code-block:: yaml
+
+   - name: configure vlans neighbor (delete others)
+     net_vlan:
+       aggregate:
+         - 1
+         - 2
+         - 3
+       state: active
+       name: reserved_vlan
+       purge: yes
+
+
+.. code-block:: yaml
+
+   - name: configure vlans neighbor (delete others)
+     net_vlan:
+       aggregate:
+         - 4
+         - { vlan_id: 5, name: mgmt }
+         - 6
+       name: reserved_vlan
+       state: active
+       purge: yes
+
+
+
+
+FIXME
+=====
+
+The following need discussing further
+
+* Should we warn if purge & not aggregate
+
+  * Do we want to add ``required_if = [('purge', 'true', ['aggregate'])]``
```

Comment:
This may not be always required. If aggregate is absent and purge is `True` it should delete all configured field under specified hierarchy.